ui/src/routes/_layout/organizations/$orgId/products/$productId/-components/product-packages-statistics-card.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { Boxes } from 'lucide-react';
+
+import { Package } from '@/api/requests';
+import { StatisticsCard } from '@/components/statistics-card';
+import { getEcosystemBackgroundColor } from '@/helpers/get-status-class';
+import { usePackagesByProductIdSuspense } from '@/hooks/use-packages-by-product-suspense';
+import { cn } from '@/lib/utils';
+import { PackageManager } from '../repositories/$repoId/-types';
+
+type ProductPackagesStatisticsCardProps = {
+  productId: number;
+  className?: string;
+};
+
+/**
+ * Calculate the counts of packages by their ecosystem.
+ *
+ * @param packages
+ * @returns Package counts sorted by ecosystem.
+ */
+const calcPackageEcosystemCounts = (
+  packages: Package[]
+): { ecosystem: PackageManager; count: number }[] => {
```

Comment:
As our types don't guarantee that `pkg.identifier.type` is `PackageManager`, I wouldn't make such assumptions here.